### PR TITLE
adding scdl dependency + proper notebook output

### PR DESCRIPTION
**Pull request by polinabinder1**
The example notebook depends on pooch, which is not installed. Additionally, the output to the scdl is written to a permanent directroy that is not deleted.

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-2.0
+0.0.4
```
Comment (jstjohn) [inline VERSION:1]: Did you mean to do this? I thought we wanted all packages to have version 2.0

**Review by jstjohn — CHANGES_REQUESTED** [review]
One maybe blocking issue, once you address I'll unblock.

File: sub-packages/bionemo-scdl/examples/example_notebook.ipynb [hunk 2]
```diff
@@ -176,13 +175,31 @@
...
+     "ename": "NameError",
+     "evalue": "name 'output_temp_dir' is not defined",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
+      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
+      "Cell \u001b[0;32mIn[11], line 2\u001b[0m\n\u001b[1;32m      1\u001b[0m dataset_temp_dir\u001b[38;5;241m.\u001b[39mcleanup()\n\u001b[0;32m----> 2\u001b[0m \u001b[43moutput_temp_dir\u001b[49m\u001b[38;5;241m.\u001b[39mcleanup()\n",
+      "\u001b[0;31mNameError\u001b[0m: name 'output_temp_dir' is not defined"
+     ]
+    }
```
Comment (skothenhill-nv) [inline sub-packages/bionemo-scdl/examples/example_notebook.ipynb:191]: I think you meant to re-run the notebook after the fix

**Review by skothenhill-nv — APPROVED** [review]
re-run the notebook, otherwise looks good.

File: VERSION (hunk 1 above)
Comment (trvachov) [reply VERSION:1]: John for scdl + bionemo-core we're going to use something like 0.0.4 and 0.0.1. Homegeneous versioning across all of bionemo is a TBD topic -- what Polina is doing here is a reversible workaround to fix SCDL (which has end users).

Here's the longer discussion: https://nvidia.slack.com/archives/C074Z808N05/p1731011971074089

and an SRD: https://docs.google.com/document/d/1Ug5fYdeo5agV3ywyO4rlgW3DDj53K8c5bgAnYNQsYK8/edit?tab=t.0#heading=h.r83g8cjg4d1y 

but that's all WIP and in the meantime SCDL needs to work.

File: sub-packages/bionemo-scdl/examples/example_notebook.ipynb [hunk 3]
```diff
@@ -91,14 +91,14 @@
...
-   "execution_count": 6,
+   "execution_count": 7,
    "metadata": {},
    "outputs": [
     {
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "/usr/local/lib/python3.10/dist-packages/bionemo/scdl/util/torch_dataloader_utils.py:39: UserWarning: Sparse CSR tensor support is in beta state. If you miss a functionality in the sparse tensor support, please submit a feature request to https://github.com/pytorch/pytorch/issues. (Triggered internally at /opt/pytorch/pytorch/aten/src/ATen/SparseCsrTensorImpl.cpp:53.)\n",
+      "/home/pbinder/.local/lib/python3.10/site-packages/bionemo/scdl/util/torch_dataloader_utils.py:39: UserWarning: Sparse CSR tensor support is in beta state. If you miss a functionality in the sparse tensor support, please submit a feature request to https://github.com/pytorch/pytorch/issues. (Triggered internally at ../aten/src/ATen/SparseCsrTensorImpl.cpp:53.)\n",
```
Comment (trvachov) [inline sub-packages/bionemo-scdl/examples/example_notebook.ipynb:101]: is it concerning that code we merge has your username in the path?

**Review by trvachov — CHANGES_REQUESTED** [review]
Fix pbinder in path?

File: VERSION (hunk 1 above)
Comment (polinabinder1) [reply VERSION]: fixed for now.

File: sub-packages/bionemo-scdl/examples/example_notebook.ipynb (hunk 2 above)
Comment (polinabinder1) [reply sub-packages/bionemo-scdl/examples/example_notebook.ipynb]: done

**Review by trvachov — APPROVED** [review]

File: sub-packages/bionemo-scdl/pyproject.toml [hunk 4]
```diff
@@ -14,6 +14,7 @@ dependencies = [
     # bionemo sub-packages
     'bionemo-core',
     # external
+    'pooch',
```
[the remark this answers is not in the document]
Comment (malcolmgreaves) [reply sub-packages/bionemo-scdl/pyproject.toml:17]: This comes in from `bionemo-core` so it's not necessary to also have it here @polinabinder1 -- https://github.com/NVIDIA/bionemo-framework/blob/main/sub-packages/bionemo-core/pyproject.toml#L24

**Review by jstjohn — APPROVED** [review]

File: sub-packages/bionemo-scdl/pyproject.toml (hunk 4 above)
[the remark this answers is not in the document]
Comment (malcolmgreaves) [reply sub-packages/bionemo-scdl/pyproject.toml:17]:
```suggestion
```

[the remark this answers is not in the document]
Comment (polinabinder1) [reply sub-packages/bionemo-scdl/pyproject.toml:17]: That can be fine. I will push a new version of bionemo.core to pypi then.

**Pull request merged**